Multisig/Features/Security/SecurityCenter.swift

```diff
@@ -163,6 +163,49 @@ class SecurityCenter {
         try changeStoreSettings(currentPlaintextPassword: oldPasscode, newPlaintextPassword: newPasscode, store: dataStore)
     }
 
+    func toggleUsage(passcodeOption: PasscodeOptions, completion: @escaping (Error?) -> Void) {
+        let enabledOptions = AppSettings.passcodeOptions
+
+        let bothTogglesWillBeOff = AppSettings.passcodeOptions.contains(passcodeOption) &&
+            AppSettings.passcodeOptions
+                .subtracting(passcodeOption)
+                .isDisjoint(with: [.useForLogin, .useForConfirmation])
+
+        if bothTogglesWillBeOff {
+            disableSecurityLock(completion: completion)
+            return
+        }
+
+        let protectionClass: ProtectionClass = passcodeOption == .useForLogin ? .data : .sensitive
+        let protectedKeyStore: ProtectedKeyStore = passcodeOption == .useForLogin ? dataStore : sensitiveStore
+
+        let toggleWillBeOff: Bool = enabledOptions.contains(passcodeOption)
+        if toggleWillBeOff {
+            requestPasswordV2(for: [protectionClass]) { [unowned self] plaintextPasscode in
+                AppSettings.passcodeOptions.remove(passcodeOption)
+                try changeStoreSettings(currentPlaintextPassword: plaintextPasscode,
+                                        newPlaintextPassword: nil,
+                                        store: protectedKeyStore)
+            } onFailure: { error in
+                AppSettings.passcodeOptions.insert(passcodeOption)
+                completion(error)
+            }
+        } else {
+            let otherProtectionClass: ProtectionClass = protectionClass == .sensitive ? .data : .sensitive
+
+            requestPasswordV2(for: [otherProtectionClass]) { [unowned self] plaintextPasscode in
+                AppSettings.passcodeOptions.insert(passcodeOption)
+                try changeStoreSettings(currentPlaintextPassword: nil,
```

Comment:
did you decide to skip the authentication in the end?

Reply:
No, I will upload latest changes